### ISSUE TITLE
Better thread naming

### ISSUE THREAD

**Issue by septag** [Celtoys/Remotery#40]
Just a suggestion ...
Instead of hashing _ThreadSampler_ pointers for thread names. It would make more sense to name threads by their IDs. something like :

```
static rmtU32 Thread_GetId()
{
#if defined(RMT_PLATFORM_WINDOWS)

    return GetCurrentThreadId();

#elif defined(RMT_PLATFORM_POSIX)

    return (rmtU32)pthread_self();

#endif
}
```

And inside `ThreadSampler_Constructor` instead of Base64_Encode line :
```
 snprintf(thread_sampler->name, sizeof(thread_sampler->name), "Thread #%d", Thread_GetId());
```

I don't know about windows, but there is also a thread naming feature in posix :
http://man7.org/linux/man-pages/man3/pthread_setname_np.3.html   
You can use that to fetch default thread name (if the user has set it before). If the name is empty, use the Thread ID as the name, if not use the name set by user.

**Comment by Celtoys** [comment]
Good idea.

**Comment by septag** [comment]
I can make a pull request for that. If you dont mind

**Comment by Celtoys** [comment]
If you can find a way of doing it without using ``snprintf`` then I'd accept a pull request for that. Additionally, as MS as deprecated even ``itoa`` for their platforms, I would prefer a custom, safe implementation to be added to Remotery to be used with the existing ``strncat_s``.

Thread names are currently a base-64 encoding of the thread sampler address as it was a convenient reuse of the ``Base64_Encode`` function.

**Comment by dougbinks** [comment]
Following on from my discovery of issue #53 I've implemented a fix and should have a pull request for this soon once I've tested on further platforms.

**Comment by dougbinks** [comment]
I have a fix for this in pull #55
